go/vt/worker/legacy_split_clone.go

```diff
@@ -50,6 +50,32 @@ import (
 	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 )
 
+type mergeStatsImpl struct {
+	m           *sync.Mutex
```

Comment:
why we need this mutex here? It was not obvious that this struct was used in a concurrent way.

Reply:
It gets passed to the row splitters; each row splitter is used per table and we spawn a bunch of them up to some concurrency limit

Reply:
caveat: this is all iirc, i don't know if i went to verify before mutexing it up but I will now out of curiosity and shame

the same mergeStatsImpl gets passed each row splitter; one row splitter is used per table and we spawn one go routine per table which shoves things to be transferred into a result channel. tl;dr the same stats struct is in a bunch of go routines maybe hitting bad rows at the same time

Reply:
Ah if it get's passed in multiple row splitters, makes sense. That was not obvious from the context of this PR.